models/client/effects.js

```diff
@@ -0,0 +1,115 @@
+module.exports = globalConfig => ({
+  messageIncoming: messageIncoming,
+  setUsername: setUsername(globalConfig),
+  sendCode: sendCode(globalConfig),
+  checkForRecovery: checkForRecovery, 
+  recover: recover(globalConfig),
+  saveLocally: saveLocally,
+  quit: quit(globalConfig)
+})
+
+function messageIncoming (state, data, send, done) {
+  var update = {
+    id: data.id
+  }
+  console.log('Message from ' + update + ' of type ' + data.type)
+  done()
+}
+
+function setUsername(globalConfig) {
+  return inner
+  function inner (state, name, send, done) {
+    if (!name) {
+      name = state.username
+    }
+    if (!name || name.length === 0) {
+      return done()
+    }
+    send('client:updateUsername', name, (err, res) => {
+      if (err) return done(err)
+      send('client:saveLocally', true, (err, res) => { if (err) done(err) })
+    })
+    if (state.connectivityState < globalConfig.connectivityStates.connected) {
+      console.log('did not publish Username, because we have no connection')
+      return done()
+    }
+    var data = {
+      type: 'USERNAME',
+      data: name
+    }
+    send('p2c:send', data, (err, res) => { if (err) done(err) })
+    done()
+  }
+
+}
+
+function sendCode(globalConfig) {
+  return inner
+  function inner (state, code, send, done) {
+    // TODO: remove, testing purpose!!
+    state.code = code
+    if (state.connectivityState < globalConfig.connectivityStates.connected) {
+      console.log('did not publish code, because no connection present')
+      return done()
+    }
+    var data = {
+      type: 'CODE',
+      data: code
+    }
+    send('p2c:send', data, (err, res) => { if (err) done(err) })
+    done()
+  }
+  
+}
+
+function checkForRecovery(state, data, send, done) {
+  send('storage:loadFromLocalStorage', null, (err, res) => {
+    if (err) return done(err)
+    if (!res) return done()
+    if (!res.id || !res.group) return done()
+    send('client:suggestRecovery', res, (err, res) => { if(err) done(err) })
+  })
+}
+
+function recover(globalConfig) {
+  return inner
```

Comment:
You are right, I was thinking of re-writing this kind of call since I use it often to get the scope for the functions right. I know there is the .bind() function with ES6 I believe, but I am really unfamiliar with it so I did not use it.
Also it may be a little bit of laziness to just nest the function this way. :/